support json output directly from Nominatim dump, without database

gcar-bendingspoons opened the pull request:
When dumping to JSON, support loading the data from a Nominatim dump instead of the database.

The goal of this change is to preprocess the JSON data dumps available from Graphhopper before importing in Komoot search service.

- [ ] Update README with new `-json <output filename> -input-file <nominatim dump filename>`
- [ ] Separate PR for adding the address type to the output

lonvia commented:
I don't understand what you want to do here. If you need to preprocess the JSON dumps you can do that with any other tool that can process json before importing them to Photon.

gcar-bendingspoons commented:
Hello @lonvia , thanks for looking into this (I still have to polish the PR though).

Initially, I noticed that the format of the JSON file provided by Graphhopper and the one resulting from running photon with `-json` are not exactly the same. We are revamping the komoot search engine that was built upon the format resulting from running photon with `-json`, and I didn't want to change the parsing too much, so I wanted to use photon directly to obtain the expected format.

Better looking into the files, I found that photon output only adds `osm_key` and `osm_value` (which I just noticed are parsed from the first value of `categories`, can you confirm), and the computed address type (which I added using the field name `type`). It also removes some other fields that we are not using.
The address type should be calculated based on the `rank_address` field.

To perform this preprocessing, from a JSON file to another, I added the possibility to bypass the database, and instead tie together the already implemented functionalities to import from JSON and dump to JSON.

Do you see an issue with adding this functionality?
